third_party/move/move-binary-format/src/errors.rs

```diff
@@ -329,7 +340,14 @@ impl PartialVMError {
         self
     }
 
-    pub fn with_message(mut self, message: String) -> Self {
+    pub fn with_message(mut self, mut message: String) -> Self {
+        if cfg!(debug_assertions)
+            && self.0.major_status == StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR
```

Comment:
I was wondering if we should just append at all times.

Reply:
I'm not sure if message goes into the block hash?

Reply:
1. We should append at all times.
2. I don't think this goes into the block hash -- only the status code does. @msmouse is this correct?